From 38d45b918ad4f7294a138a83d26c87635bd4bcfd Mon Sep 17 00:00:00 2001
From: Gustavo Ornaghi Antunes <gustavo.ornaghiantunes@windriver.com>
Date: Mon, 26 Feb 2024 10:15:38 -0300
Subject: [PATCH] csi-powerstore: add secret in templates

This patch consists of adding a secret in the helm chart templates directory.
It's needed because this helm chart needs a config secret that should be
created manually.

Signed-off-by: Gustavo Ornaghi Antunes <gustavo.ornaghiantunes@windriver.com>
---
 charts/csi-powerstore/templates/secret.yaml | 21 +++++++++++++++++++++
 1 file changed, 21 insertions(+)
 create mode 100644 charts/csi-powerstore/templates/secret.yaml

diff --git a/charts/csi-powerstore/templates/secret.yaml b/charts/csi-powerstore/templates/secret.yaml
new file mode 100644
index 0000000..40ab5c6
--- /dev/null
+++ b/charts/csi-powerstore/templates/secret.yaml
@@ -0,0 +1,21 @@
+{{- if .Values.secret }}
+apiVersion: v1
+kind: Secret
+metadata:
+  name: {{ .Release.Name }}-config
+  namespace: {{ .Release.Namespace }}
+stringData:
+  config: |
+    arrays:
+      {{- range .Values.secret.arrays }}
+      - globalID: "{{ .globalID }}"
+        username: "{{ .username }}"
+        password: "{{ .password }}"
+        endpoint: "{{ .endpoint }}"
+        skipCertificateValidation: {{ .skipCertificateValidation | default "true" }}
+        isDefault: {{ .isDefault | default "false" }}
+        blockProtocol: {{ .blockProtocol | default "auto" }}
+        nasName: "{{ .nasName }}"
+        nfsAcls: "{{ .nfsAcls | default "0777" }}"
+      {{- end }}
+{{- end }}
-- 
2.34.1
